### DISCUSSION TITLE
SubColumns, different from column grouping

### DISCUSSION THREAD

**Discussion by ingegneri-bruno-clorox** [KevinVandy/material-react-table#934] [Ideas]
I'm creating a table where I want to give the user the ability to group/ungroup columns by period, but keep them as columns and not as rows.

What I want is this similar look as we got on Excel.
![Screenshot 2024-01-10 at 18 49 21](https://github.com/KevinVandy/material-react-table/assets/116366661/f585fa00-1df0-4356-9414-b7bdf139b6f7)

But what MRT gives it's this grouping that turns columns into "rows", which is not ideal.
![Screenshot 2024-01-10 at 18 53 49](https://github.com/KevinVandy/material-react-table/assets/116366661/ce0bebf2-482c-4373-b003-26046576403b)

Not sure if there's some way of doing that, but if not, that would be a nice update for it.

**Comment by KevinVandy** [comment]
We call those header groups in TanStack Table, and yeah definitely supported as seen in the advanced example:
https://www.material-react-table.com/docs/examples/advanced

**Reply by ingegneri-bruno-clorox** [reply]
HI, @KevinVandy. Thanks for the reply. It gave me another idea that I'll try and update the thread if succeeded.

Before posting here, I was able to use header groups and also created a custom expansion that hide columns onClick, but it was not trivial.

Just ideas, not sure if it's feasible. But it would be nice if MRT header grouping had optional functionalities, like:

1. Expand/collapse "subColumns"

2. Automatically generated **aggregation column** (such as sum or average of totals).
_On my solution, I had to calculate total before and send data as a "subColumn" extra field._

3. Visual hints, just like what we see on EXCEL outline grouping.
![Screenshot 2024-01-11 at 10 26 57](https://github.com/KevinVandy/material-react-table/assets/116366661/ba1676c0-78e0-4124-af9d-f1e46255ca67)
